optuna/integration/pytorch_distributed.py

```diff
@@ -278,7 +278,9 @@ def set_system_attr(self, key: str, value: Any) -> None:
         if dist.get_rank(self._group) == 0:
             try:
                 assert self._delegate is not None
-                self._delegate.storage.set_trial_system_attr(self._delegate._trial_id, key, value)
+                self._delegate.storage.set_trial_system_attr(  # type: ignore[attr-defined]
```

Comment:
> I found that BaseTrial does not have an attribute .storage and trial_id, thus I needed to add type: ignore to some lines.

As you mentioned, `BaseTrial` does not have `storage` attribute, and the code will fail here. If we want to support this method with `FrozenTrial`, we will need to consider implementing additional changes that allow functionality without relying on the storage attribute.

Regarding `TorchDistributedTrial`, it's not required for it to fully support `BaseTrial` in this PR. What do you think about adding a TODO comment as a reminder for us to revisit this issue in future iterations of the code?

In addition, if we clarify that the current implementation only supports for `Trial` class, we can check `isinstance(self._delegate, Trial)`  at L280 and remove `type: ignore comment` at L281 and L282.

Reply:
@toshihikoyanase Since `set_system_attr` has been deprecated, how about just adding a comment that it does not work with `FrozenTrial` instead of TODO?

Reply:
@not522 Sounds reasonable. Thank you for your suggestion. We can add the comment in a follow-up PR so I examine the current change.